docs/joss/paper.md

```diff
@@ -0,0 +1,203 @@
+experience and productivity.
+
+The user interface of SpeedyWeather.jl is heavily influenced by
+the Julia ocean model Oceananigans.jl [@Ramadhan2020].
+A monolithic interface based on parameter files is avoided in favor of a
+library-style interface in which users write code to run models rather than
+merely supplying parameters and input arrays.
+A model is created bottom-up by first defining the discretization
+and any non-default model components with its respective parameters.
+All components are then collected into a single model object, which, once
+initialized, returns a simulation object that contains the entire model state,
+work arrays and parameters, that can be run, analysed or changed.
+While these steps can be written into a script for reproducibility,
+the same steps can be executed and interacted with one-by-one in
+Julia's read-evaluate-print loop (REPL). We thereby reach an interactivity
+far beyond a monolithic interface that is limited to the options provided.
+At the same time, sensible default arguments enable even inexperienced users
+to run simulations in just a few lines of code. 
+
+To be extensible and composable with new
+model components, SpeedyWeather.jl relies on Julia's multiple dispatch
+programming paradigm [@Bezanson2017]. Every model component
+is defined as a new type. For example, to define a new way how to calculate
+the precipitation due to the physical process of large-scale condensation,
+one would define `MyCondensation` as a new subtype of `AbstractCondensation`.
+One then only needs to extend the `initialize!` and `condensation!`
+functions for this new type. Passing on `condensation = MyCondensation()`
+to the model constructor then implements this new model component without
+the need to branch off or overwrite existing model components.
+Conceptually similar scientific modelling paradigms have been very successful
+in the Python-based generic partial differential equation solver Dedalus [@Burns2020]
+and the Julia ocean model Oceananigans.jl [@Ramadhan2020].
+
+![Surface temperature simulated with the primitive equation model in SpeedyWeather.jl.
+(Figure will be updated) \label{fig:primitive}](primitive.jpg)
+
+The dynamical core of SpeedyWeather.jl uses established numerics
+[@Bourke1972; @Hoskins1975; @Simmons1978; @Simmons1981],
+widely adopted in numerical weather prediction. It is based on the spherical
+harmonic transform with a leapfrog-based semi-implicit time integration [@Hoskins1975]
+and a Robert-Asselin-Williams filter [@Williams2011; @Amezcua2011].
+The spherical harmonic transform is grid-flexible. Any iso-latitude ring-based
+grid can be used and new grids can be externally defined and passed in
+as an argument. Many grids are already implemented: the conventional
+Gaussian grid, a regular longitude-latitude grid, 
+the octahedral Gaussian grid [@Malardel2016], the octahedral
+Clenshaw-Curtis grid [@Hotta2018], and the HEALPix grid [@Gorski2005].
```

Comment:
Similar as before, I don't see the win over the new paragraph compared to the old, especially because we now have two very long sentences, whereas the previous version had a clear summary sentence to start with before going into the details.